### PR TITLE
Chunk: Use enum class instead of boolean parameter

### PR DESCRIPTION
**Pull request by mrzzzrm**
Because noone likes to read code with boolean params. Also, Chunk had too many constructors.

### CODE REVIEW

File: src/lib/import_export/csv_parser.cpp [hunk 1]
```diff
@@ -40,7 +40,7 @@ std::shared_ptr<Table> CsvParser::parse(const std::string& filename) {
   std::vector<size_t> field_ends;
   while (_find_fields_in_chunk(content_view, *table.get(), field_ends)) {
     // create empty chunk
-    chunks.emplace_back(true);
```
Comment (mjendruk) [inline src/lib/import_export/csv_parser.cpp]: This is a good example against the default usage of `emplace_back`. Although `explicit Chunk(const bool has_mvcc_columns);` is marked explicit, `emplace_back` implicitly converts a boolean value to a `Chunk`. Our linter even complains about constructors with a single parameter yet we circumvent it by our always-emplace-back policy.

Could please replace `emplace_back` with `push_back`? :)

File: src/lib/storage/chunk.hpp [hunk 2]
```diff
@@ -18,10 +18,17 @@ namespace opossum {
 class BaseIndex;
 class BaseColumn;
 
-// A chunk is a horizontal partition of a table.
-// It stores the data column by column.
-//
-// Find more information about this in our wiki: https://github.com/hyrise/zweirise/wiki/chunk-concept
+enum class ChunkMvccMode {
+  UseMvccColumns, NoMvccColumns
```
Comment (mjendruk) [inline src/lib/storage/chunk.hpp]: Two suggestions:
```c++
enum class ChunkMvccMode { On, Off };
```
```c++
enum class ChunkCreateMvccColumns { Yes, No };
```

Reasoning: it’s shorter.

**Comment by mrks** [comment]
This would be the perfect place for [P0329R4](http://wg21.link/p0329r4) if only clang was keeping up... ;)

**Comment by mrks** [comment]
Since I feel like this is introducing a new code style, I would like to keep this open for a while and see if anyone has anything to say about it.

File: src/lib/import_export/csv_parser.cpp (hunk 1 above)
Comment (mrzzzrm) [reply src/lib/import_export/csv_parser.cpp]: I decided (and Max didn't fight back) that this is an appropriate usage on `emplace_back`

**Comment by mrzzzrm** [comment]
`reinterpret_cast<Elephant*>(midge_ptr)`.

Imo this is not that much of a new code style nor is the grammar off. I'd just merge it.

**Pull request merged**